pkg/util/cert.go

```diff
@@ -15,3 +33,49 @@ func EncodeCertsPEM(cert *x509.Certificate, caCerts []*x509.Certificate) []byte
 	}
 	return pemBytes
 }
+
+// GetCertUsages returns a slice of strings representing the certificate usages
+func GetCertUsages(cert *x509.Certificate) []string {
+	usages := []string{}
+	if cert.KeyUsage&x509.KeyUsageCertSign != 0 {
+		usages = append(usages, CertUsageCertSign)
+	}
+	for _, eku := range cert.ExtKeyUsage {
+		switch eku {
+		case x509.ExtKeyUsageServerAuth:
+			usages = append(usages, CertUsageServerAuth)
+		case x509.ExtKeyUsageClientAuth:
+			usages = append(usages, CertUsageClientAuth)
+		}
+	}
+	if len(usages) == 0 {
+		usages = append(usages, CertUsageUnknown)
+	}
+	return usages
+}
+
+// GetCertStatus determines the status of a certificate based on its validity period
+func GetCertStatus(cert *x509.Certificate, now time.Time, warn time.Time) string {
+	if now.Before(cert.NotBefore) {
+		return CertStatusNotYetValid
+	} else if now.After(cert.NotAfter) {
+		return CertStatusExpired
+	} else if warn.After(cert.NotAfter) {
+		return CertStatusWarning
+	}
+	return CertStatusOK
+}
+
+// FormatDuration formats a duration in a human-readable format (years, days, or hours)
+func FormatDuration(d time.Duration) string {
+	if d.Hours() >= 24*365 {
+		years := int(d.Hours() / (24 * 365))
+		return fmt.Sprintf("%dy", years)
+	}
+	if d.Hours() >= 24 {
+		days := int(d.Hours() / 24)
+		return fmt.Sprintf("%dd", days)
+	}
+	hours := int(d.Hours())
+	return fmt.Sprintf("%dh", hours)
+}
```

Comment:
curious why this instead of just the precise output of `duration.String()`?

Reply:
`FormatDuration` initially matched Kubeadm's style but had precision and edge case issues. 
I've removed it and now use `duration.String()`, which is more mature and stable.

Reply:
If you wanted to improve on this you could use `github.com/dustin/go-humanize`. We already have an indirect dep on it by way of etcd, so it wouldn't be net new.

Reply:
Fixed, thanks! 👍